runtime/compiler/compile/J9Compilation.cpp

```diff
@@ -341,12 +341,12 @@ J9::Compilation::printCompYieldStats()
 const char *
 J9::Compilation::getContextName(TR_CallingContext context)
    {
-   if (context == OMR::endOpts || context == TR_CallingContext::NO_CONTEXT)
+   if ((OMR::Optimizations)context == OMR::endOpts || context == TR_CallingContext::NO_CONTEXT)
       return "NO CONTEXT";
-   else if (context < OMR::numOpts)
+   else if ((OMR::Optimizations)context < OMR::numOpts)
```

Comment:
It's extremely unlikely to cause a problem, but the range of `OMR::Optimizations` could be different than the range of `TR_CallingContext`.  One could have a range of [0..255], say, while the other has a range of [0..127].  If the value of `context` was outside the range of `OMR::Optimization`, casting to it would result in undefined behaviour.

It might be safer to cast `OMR::endOpts` and `OMR::numOpts` to `TR_CallingContext` rather than casting `context` to `OMR::Optimizations` - the way `TR_CallingContext` and `OMR::Optimization` are defined, `endOpts` and `numOpts` are always in the range of `TR_CallingContext`.